lib/Sema/MiscDiagnostics.cpp

```diff
@@ -2093,7 +2093,11 @@ class VarDeclUsageChecker : public ASTWalker {
   /// This is a mapping from an OpaqueValue to the expression that initialized
   /// it.
   llvm::SmallDenseMap<OpaqueValueExpr*, Expr*> OpaqueValueMap;
-  
+
+  /// This is a mapping from a VarDecl property getter to the expression that
+  /// used it.
+  llvm::SmallDenseMap<VarDecl*, DeclRefExpr*> GetterDeclUsageMap;
```

Comment:
This map will only ever have one entry, right?

Reply:
Yea -- sort of funny, I woke up this morning and that was literally my first thought. I converted it to a pair, but I'm not sure if that's a normal C++ construct to use. Would that make sense to you?

Reply:
I think I would just use two instance variables, and not worry about keeping them tied together. You also set each half at a different point in execution anyway.

(Also, may as well make them `const`.)